### PR TITLE
Improve mid-combat roleplay code.

### PR DESCRIPTION
**Pull request by smartin015**
- Now allows for `**win**` and `**lose**` trigger elements in QDL to trigger the win and lose events. This is a departure from the previous behavior, where `**goto ID**` would show the win or lose screen when the target was within the win or lose event of combat.
- Actually stop combat audio on edge cases (e.g. `**end**` trigger within combat RP)
- Fixes `yarn app-watch-test` so it actually rebuilds and executes the tests on file change
- Add some safety code to stop proliferation of "unknown event win/lose" at the end of combat. If we reach the victory card with a parsernode that's a roleplay node *within* the combat node, it correctly chooses the next node based on the parent combat node.
- resolve phase was improperly setting quest node before taking a history snapshot, which may have contributed to bugs.

#399

### CODE REVIEW

File: services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx [hunk 1]
```diff
@@ -172,20 +173,20 @@ describe('Combat actions', () => {
 
     it('randomly assigns damage', () => {
       const {actions} = newStore({});
-      expect(actions[3].node.ctx.templates.combat.mostRecentAttack.damage).toBeDefined();
+      console.log(actions);
```
Comment (toddmedema) [inline services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx]: rm console

File: services/app/karma.conf.js [hunk 2]
```diff
@@ -1,31 +1,30 @@
...
-const webpack = require('webpack');
 
 webpackConfig.module.rules.unshift({
   test: /isIterable/,
   loader: 'imports?Symbol=>false'
 });
+webpackConfig.entry = undefined;
+
+// Remove hot reload
+webpackConfig.plugins = [webpackConfig.plugins[webpackConfig.plugins.length-1]];
```
Comment (toddmedema) [inline services/app/karma.conf.js]: Would it be possible to be more principled about this, ie to explicitly remove the hot reload plugin rather than assume it's position?

File: services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx [hunk 3]
```diff
@@ -239,6 +240,32 @@ describe('Combat actions', () => {
 
       expect(store.getActions()[1].node.ctx.templates.combat.levelUp).toEqual(false);
     });
+
+    it('Fixes when given a mid-combat roleplay node', () => {
```
Comment (toddmedema) [inline services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx]: This test title feels unclear what it's actually testing?

File: services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx (hunk 1 above)
Comment (smartin015) [reply services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx]: Done

File: services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx (hunk 3 above)
Comment (smartin015) [reply services/app/src/components/views/quest/cardtemplates/combat/Actions.test.tsx]: Done.

File: services/app/karma.conf.js (hunk 2 above)
Comment (smartin015) [reply services/app/karma.conf.js]: Made an attempt at this :) hard to do when the copy plugin has poor naming though.
